### PR TITLE
#2811 setup onboarding progress bar

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

- progress bar for checklists
- overall progress
- finish button

## Screenshots
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #2811)

### CODE REVIEW

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Can you make the progress bar look a little more like the mocks and also for the checklists, i think the progress bar should be next to the text

File: src/frontend/src/components/OnboardingProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Box, LinearProgress, linearProgressClasses, styled, SxProps, Typography } from '@mui/material';
+
+interface ProgressBarWithValueProps {
```
Comment (walker-sean) [inline src/frontend/src/components/OnboardingProgressBar.tsx]: Please rename to `OnboardingProgressBarProps`

**Review by walker-sean — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/HomePage/OnboardingHomePage.tsx [hunk 2]
```diff
@@ -21,14 +24,11 @@ const OnboardingHomePage = () => {
...
-      <Grid container display={'flex'} alignItems={'center'} marginLeft={2} marginTop={4}>
+      <Grid container display="flex" alignItems="center" justifyContent={'space-between'} padding={1} marginTop={4}>
         <Grid item xs={12} md={7}>
           <Typography sx={{ fontSize: '2.5em' }}>Welcome to the {organization.name} Team</Typography>
         </Grid>
-        {/* This will be replaced with the 'Finished' button*/}
-        <Grid item xs={12} md={5} sx={{ display: 'flex', justifyContent: 'flex-end' }}>
-          <Typography sx={{ fontSize: '2em' }}>Finished</Typography>
-        </Grid>
+        <NERButton variant="contained">Finished?</NERButton>
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/HomePage/OnboardingHomePage.tsx:31]: this button doesnt do anything?

File: src/frontend/src/pages/HomePage/components/Checklist.tsx [hunk 3]
```diff
@@ -16,19 +17,23 @@ const Checklist: React.FC<{ parentChecklists: ChecklistType[]; checklistName?: s
...
-          <Grid display="flex" alignItems="center" justifyContent="space-between">
-            <Typography fontSize="2em" fontWeight="bold">
-              {checklistName} Checklist
-            </Typography>
-            <Grid display="flex" alignItems="center" gap={2}>
-              <progress value={50} max={100} />
+    <Box sx={{ backgroundColor: theme.palette.background.paper, borderRadius: 5, p: 2 }}>
+      <Grid container spacing={2}>
+        <Grid item xs={12}>
+          <Grid container justifyContent="space-between" alignItems="center">
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/HomePage/components/Checklist.tsx]: whats going on with all these grids. Are these really all necessary

**Review by Peyton-McKee — COMMENTED** [review]
You can merge if these comments dont uncover anything

**Pull request merged**